gc/2022/GC-08-23.md

```diff
@@ -33,3 +33,187 @@ Installation is required, see the calendar invite.
+
+AR: there’s never a point to be sure, over time we might have more data about what size data is more relevant, personally I think 5% is not an issue but ymmv, but we will be able to gather more data on that.  The other side is that some features that are at higher risk, we will have added already,  we will know more about whether these risks materialize, there will always be potential more things there are very likely features that we have them, type parameters for example, once we have those I’d be somewhat more comfortablle
+
+JK: Having annotations make some tools simpler, don’t think that’s a convincing argument. In particular a bunch of arguments  regarding tools don’t make much sense, if you want to look for all occurrences of something, then you have to look everywhere anyway because you can reconstruct the types in a single linear pass over the module, and you have to do a single linear pass over the module if you want to look for all occurrences of something, that’s not really a significant burden, it only affects tools that process fragments of Wasm functions. Do we really want to make regular production modules that ship to billions of users larger by x%, because there could be a tool that wants to analyze fragments, and that tool would be slightly simpler? I remain extremely unconvinced by that line of reasoning.
+
+BT: We are designing a format that has many different purposes, it’s a code format for interchange, we should prioritize all the tools that process that format, producers and consumers, and I don’t see how we justify that.. We could make the reverse argument, why aren’t there annotations that would be useful for tool authors and engines? From a different point of view that might be a hostile design decision
+
+AR: This is very real feedback from people that work in the research space, one of the top things that makes their life easier with Wasm was that there is no overload, which is the trivial case of the problem we’re discussing right now. We’re talking about a much more inconvenient relaxation.. I would also like to push back on the size arguments, it falls into the category of premature optimization, we haven’t really optimized for size as much previously, putting in adhoc solutions to individual small things, we could have had a const 0 inaructions, but we decided against it, we don’t have a negate instructions, and all these things we never decided to size optimized, we should do a more targeted approach instead of adhoc ideas, we should stay consistent with that
```

Comment:
```suggestion
AR: This is very real feedback from people that work in the research space, one of the top things that makes their life easier with Wasm was that there is no overload, which is the trivial case of the problem we’re discussing right now. We’re talking about a much more inconvenient relaxation. I would also like to push back on the size arguments, it falls into the category of premature optimization, we haven’t really optimized for size as much previously, putting in ad hoc solutions to individual small things, we could have had a const 0 instruction, but we decided against it, we don’t have a negate instruction, and all these things we never decided to size optimize, we should do a more targeted approach instead of ad hoc ideas, we should stay consistent with that.
```